### PR TITLE
Automation and gherkin for Serverless Usability Improvements

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-6359
Story: https://issues.redhat.com/browse/ODC-6420

Acceptance criteria:

- Add Subscription should enable any service to be added as a Subscriber
- Add Trigger should enable any service to be added as a Subscriber
- Create Channel form should provide the ability to switch back & forth between YAML/Form without losing data
- 2 additional Serverless dashboards should be available in Observe Dashboard in the Dev Perspective

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6422

**Test Setup**:

Check the TAGS in frontend/packages/knative-plugin/integration-tests/cypress.json file be: """(@knative or @knative-eventing) and (@pre-condition or @smoke or @regression) and @[odc-6359](https://issues.redhat.com/browse/odc-6359) and not (@manual or @to-do or @un-verified or @broken-test)""",

Check the TAGS in frontend/packages/dev-console/integration-tests/cypress.json file be: ""(@pre-condition or @smoke or @regression) and @[odc-6359](https://issues.redhat.com/browse/odc-6359) and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `npm run test-cypress-knative` and `npm run test-cypress-dev-console` simultaneously

Execute the create-event-sources.feature, eventing-channel-subscription.feature, broker-trigger.feature and in dev-console dashboard-display.feature file

**Browser**
Chrome 90

**Test Execution Screenshot:**

### CODE REVIEW

**Review by jerolimov — APPROVED** [review]
I'm fine with the changes, @sahil143 can you please check if the feature files match the epic changes and AC? Thank you.

/cc @sahil143

File: frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx [hunk 1]
```diff
@@ -56,12 +56,13 @@ const ApiServerSection: React.FC<ApiServerSectionProps> = ({ title, fullWidth })
...
   };
   return (
-    <FormSection title={title} extraMargin fullWidth={fullWidth}>
+    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title}-section`}>
       <FormGroup
         fieldId={fieldId}
         label={t('knative-plugin~Resource')}
         helperText={t('knative-plugin~The list of resources to watch')}
         isRequired
+        data-test="Resource-group"
```
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx:65]: Unused :)

File: frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx [hunk 2]
```diff
@@ -56,12 +56,13 @@ const ApiServerSection: React.FC<ApiServerSectionProps> = ({ title, fullWidth })
     }
   };
   return (
-    <FormSection title={title} extraMargin fullWidth={fullWidth}>
+    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title}-section`}>
```
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx:59]: You maybe know, I would always recommend separate title and section with a space to enable search by `~=`.

```suggestion
    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title} section`}>
```

File: frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx]: Removed

File: frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/knative-plugin/src/components/add/event-sources/ApiServerSection.tsx]: Updated

File: frontend/packages/knative-plugin/src/components/add/event-sources/ContainerSourceSection.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ const ContainerSourceSection: React.FC<ContainerSourceSectionProps> = ({ title,
     [setFieldValue],
   );
   return (
-    <FormSection title={title} extraMargin fullWidth={fullWidth}>
+    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title}-section`}>
```
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/components/add/event-sources/ContainerSourceSection.tsx:54]: If you accept the spaces you need to change this here as well
```suggestion
    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title} section`}>
```

File: frontend/packages/knative-plugin/src/components/add/event-sources/SinkBindingSection.tsx [hunk 4]
```diff
@@ -61,7 +61,7 @@ const SinkBindingSection: React.FC<SinkBindingSectionProps> = ({ title, fullWidt
   }, [matchType, setFieldValue]);
 
   return (
-    <FormSection title={title} extraMargin fullWidth={fullWidth}>
+    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title}-section`}>
```
Comment (jerolimov) [inline frontend/packages/knative-plugin/src/components/add/event-sources/SinkBindingSection.tsx:64]: And here
```suggestion
    <FormSection title={title} extraMargin fullWidth={fullWidth} dataTest={`${title} section`}>
```

**Review by vikram-raj — COMMENTED** [review]
Tried it locally and all tests are passing.

/lgtm
/label qe-approved

**Comment by sanketpathak** [comment]
This PR only affects our e2e tests.
/label px-approved
/label docs-approved

**Pull request merged**